Userland/Services/WindowServer/WindowFrame.cpp

```diff
@@ -966,4 +973,37 @@ int WindowFrame::menu_row_count() const
     return m_window.menubar().has_menus() ? 1 : 0;
 }
 
+void WindowFrame::latch_window_to_screen_edge(ResizeDirection resize_direction)
+{
+    auto window_rect = m_window.rect();
+    auto frame_rect = rect();
+    auto& screen = Screen::closest_to_rect(window_rect);
+    auto screen_rect = screen.rect();
+
+    if (screen.is_main_screen())
+        screen_rect.shrink(0, 0, TaskbarWindow::taskbar_height(), 0);
```

Comment:
:thonk: Might be nice to have a function somewhere for this, so that user code like this doesn't have to care which side of the screen the taskbar is on, it can just say "gimme the usable screen area".

Though, we don't support moving the taskbar (or having an extra bar for the global menu) so it doesn't actually matter here.

Reply:
Derp, I just found this actually does exist. `WindowManager::the().desktop_rect(screen)`

At least, I think that's the same.

Reply:
Yes that's the same! Good find. I copied this logic over from the window tiling logic, so that should be updated there as well.

Reply:
Cool, I'll get on that